### PR TITLE
SPEED-1: Add the Kafka Producer Model to send status updates to the leader

### PR DESCRIPTION
**Pull request by jakejack13**
### Description

This PR creates a new output stream for sending results of tests to the leader. We also create the start of the leader

### CODE REVIEW

File: leaders/src/main/java/edu/cornell/Main.java
Comment (MitchellGray100) [inline leaders/src/main/java/edu/cornell/Main.java:12]: I would make these static constants in a separate file

File: workers/src/main/java/edu/cornell/repository/JUnit5RepositoryImpl.java
Comment (MitchellGray100) [inline workers/src/main/java/edu/cornell/repository/JUnit5RepositoryImpl.java:24]: is this TODO supposed to be left in?

File: workers/src/main/java/edu/cornell/repository/JUnit5RepositoryImpl.java
Comment (MitchellGray100) [inline workers/src/main/java/edu/cornell/repository/JUnit5RepositoryImpl.java:25]: Or this FIXME

**Review by MitchellGray100 — CHANGES_REQUESTED** [review]
See comments then LGTM

File: leaders/src/main/java/edu/cornell/Main.java
Comment (jakejack13) [reply leaders/src/main/java/edu/cornell/Main.java:12]: When we spin up a new leader (because each pipeline will use one leader for coordination, so multiple leaders), we need to pass these inputs into that leader. It might be difficult to have these in a config file since the leader won't know where that config file is. I think env variables are easier because they can be passed into the Docker container on startup

File: workers/src/main/java/edu/cornell/repository/JUnit5RepositoryImpl.java
Comment (jakejack13) [reply workers/src/main/java/edu/cornell/repository/JUnit5RepositoryImpl.java:24]: Should be removed once Owen merges his stuff. I'm going to keep it in just in case something goes wrong during the merge and I can remember where things need to be hooked up

File: workers/src/main/java/edu/cornell/repository/JUnit5RepositoryImpl.java
Comment (jakejack13) [reply workers/src/main/java/edu/cornell/repository/JUnit5RepositoryImpl.java:25]: Good call. This was left in for testing. I'll go ahead and remove it now

**Review by MitchellGray100 — APPROVED** [review]
LGTM when CI/CD passes

**Review by owenxr — APPROVED** [review]
LGTM

**Pull request merged**
